src/middlewares/logging.ts

```diff
@@ -0,0 +1,68 @@
+import express from "express";
+
+class Logging {
+	RequestResponseLogger = (data: any, req:express.Request, res:express.Response, _next:express.NextFunction) => {
+		this.ClearResponseBeforeSend(res);
+		if (data.success) {
+			const log = this.CreateLog(JSON.stringify(data), null, req, res);
+			console.log(log);
+			//TODO: do something with log. write it in a db or maybe send an elastic search instance
+
+			const statusCode = data.statusCode;
+			delete data.statusCode;
+			res.status(statusCode).json(data);
+		} else {
+			if (data instanceof Error) {
+				const defaultErrorResponse = {success:false, message:"Internal Server Error"};
+				const log = this.CreateLog(JSON.stringify(defaultErrorResponse), data, req, res);
+				console.log(log);
+				//TODO: do something with log. write it in a db or maybe send an elastic search instance
+
+				res.status(500).json(defaultErrorResponse);
+			} else {
+				const log = this.CreateLog(JSON.stringify(data), `${data.err} => ${data.err.stack}`, req, res);
+				console.log(log);
+				//TODO: do something with log. write it in a db or maybe send an elastic search instance
+
+
+				res.status(data.statusCode).json(data);
+			}
+		}
+	};
+
+	private CreateLog = (content: any, error: any, req:express.Request, res:express.Response) => {
+		return {
```

Comment:
This becomes JSON output? Since this is not an explicit type, can we use `snake_case` in property name please?